### PR TITLE
Fix scaling of spectrum

### LINKED ISSUE #73: `scale_spectrum()` is used incorrectly and therefore does not do anything
This line has no effect:

https://github.com/AstarVienna/ScopeSim_Templates/blame/25c262ff6da70645655458db1b0183e9a871f442/scopesim_templates/misc/misc.py#L199

```python
    tu.scale_spectrum(spec, filter_name=filter_name, amplitude=amp_unit)
```

`scale_spectrum()` essentially does
```
def scale_spectrum(spectrum, scale_factor)
    spectrum *= scale_factor.value
    return spectrum
```

where `spectrum` is a `synphot.SourceSpectrum`.

The problem is that `SourceSpectrum`  does not  define `__rmul__`, so the multiplication line is not an in-place multiplication. That is, the `spectrum` that is given as input, in this case `spec` is not modified. Instead, a new `SourceSpectrum` is created, which is returned and subsequently discarded.

So the line at the top should be
```python
    spec = tu.scale_spectrum(spec, filter_name=filter_name, amplitude=amp_unit)
```

And for clearity, we should perhaps not use `*=` here, since that seems to imply an in-place modification. Instead, perhaps we should just do
```
    spectrum = spectrum * scale_factor.value
```

### PR DESCRIPTION
**Pull request by hugobuddel**
Fixes #73

### CODE REVIEW

**Comment by hugobuddel** [comment]
Perhaps we should fix the bug that requires synphot templates to require 4 or more wavelength points instead of these kludgey workarounds.

The tests I added here do not give any flux when only 2 or 3 points are used. I think it might be a problem with determining overlaps. Not sure.

File: docs/starting.rst [hunk 1]
```diff
@@ -28,8 +28,8 @@ user
     hdr = fits.Header(dict(NAXIS=2,
                            NAXIS1=img.shape[0]+1,
                            NAXIS2=img.shape[1]+1,
-                           CRPIX1=0,
-                           CRPIX2=0,
+                           CRPIX1=img.shape[0]/2,
+                           CRPIX2=img.shape[1]/2,
```
Comment (teutoburg) [inline docs/starting.rst:32]:
```suggestion
                           CRPIX1=(img.shape[0] + 1) / 2,
                           CRPIX2=(img.shape[1] + 1) / 2,
```

File: scopesim_templates/tests/pyobjects/source_objects.py [hunk 2]
```diff
@@ -26,6 +28,34 @@ def _basic_imagehdu(n=11):
...
+    sigma = 5
+    x, y = np.meshgrid(np.arange(n), np.arange(n))
+    img = np.exp(-1 * (((x - n / 2) / sigma) ** 2 + ((y - n / 2) / sigma) ** 2))
+
+    # Fits headers of the image. Yes it needs a WCS
+    hdr = fits.Header(dict(NAXIS=2,
+                           NAXIS1=n + 1,
+                           NAXIS2=n + 1,
+                           CRPIX1=n / 2,
+                           CRPIX2=n / 2,
```
Comment (teutoburg) [inline scopesim_templates/tests/pyobjects/source_objects.py:44]:
```suggestion
                           NAXIS1=n,
                           NAXIS2=n,
                           CRPIX1=(n + 1) / 2,
                           CRPIX2=(n + 1) / 2,
```

**Review by teutoburg — COMMENTED** [review]
I might be missing something, but I think these (see other comments) would end up "off-by-0.5-1.0" otherwise. But I can't really double check that right now (on the train). Everything else should be good 👍

**Comment by hugobuddel** [comment]
Thanks for catching that possible off-by-0.5 error. The source is still far from the center in the simulated images though... The peak value (so ~center) seems to be at 2123,2123 instead of 2048,2048.

Nevertheless, the exact position of the source doesn't really matter here, as it is a test for the photometry, not the astrometry. The source should just not be entirely at the corner. So I decided to not investigate that offset further.

I created #75 to fix the problem of requiring 4 or more points for a synphot spectrum (or all spectra?), as it is also essentially independent of this problem. Or well, it did make scaling of the spectra in `source_from_imagehdu` broken, but I 'fixed' that by simply using enough points.

So I prefer to merge this first, and since you said 'everything else is good', I'll do just that :-).

Also, I turned the figures off by default, oops.

**Pull request merged**